### DISCUSSION TITLE
Shouldn't this icon colour override work?

### DISCUSSION THREAD

**Discussion by mitya33** [tusen-ai/naive-ui#4882] [General]
Shouldn't the below set the colour of the icon to red, or have I misunderstood?

```
<template>
    <n-icon size='medium' :theme-overrides='foo' :component='CheckCircleRegular' />
</template>
<script setup>
import { CheckCircleRegular } from '@vicons/fa'
const foo = {color: 'red'};
</script>
```

Also tried:

```
const foo = {Icon: {color: 'red'}};
```

Icon colour doesn't change. Thanks in advance.

**Comment by jahnli** [comment]
```
<n-icon size='medium' :color='red' :component='CheckCircleRegular' />
```

**Reply by mitya33** [reply]
Thanks, that works. But just out of interest, shouldn't my attempt have worked too? The theme explorer confirms that the `Icon` component supports a `color` CSS property, so I thought I could set that via a theme overrides parent.

**Comment by jahnli** [comment]
`theme-overrides` act on `n-config-provider`

you can try

```
const themeOverrides: GlobalThemeOverrides = {
  Icon: {
    color: 'red'
  }
};

<n-config-provider
  inline-theme-disabled
  :theme-overrides="themeOverrides"
>
  <n-loading-bar-provider>
    <n-dialog-provider>
      <n-notification-provider>
        <n-message-provider>
          <app />
        </n-message-provider>
      </n-notification-provider>
    </n-dialog-provider>
  </n-loading-bar-provider>
</n-config-provider>
```

**Reply by mitya33** [reply]
Thanks, I'll try that.

**Reply by jahnli** [reply]
@mitya33  For the discussion, if you feel that the answer to a question is correct, hopefully you can mark it as answered, which will help others who are interested in the question

**Reply by mitya33** [reply]
Thanks for the heads up. Will do.
